pkg/blockstorage/azure/auth.go

```diff
@@ -1,110 +1,167 @@
 package azure
 
 import (
-	"context"
-
-	"github.com/Azure/go-autorest/autorest/adal"
-	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
 	"github.com/kanisterio/kanister/pkg/blockstorage"
 	"github.com/pkg/errors"
 )
 
+const ActiveDirectory = "activeDirectory"
+
 // currently avaialble types: https://docs.microsoft.com/en-us/azure/developer/go/azure-sdk-authorization
 // to be available with azidentity: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity#readme-credential-types
 // determine if the combination of creds are client secret creds
 func isClientCredsAvailable(config map[string]string) bool {
-	return (config[blockstorage.AzureTenantID] != "" &&
+	return config[blockstorage.AzureTenantID] != "" &&
 		config[blockstorage.AzureClientID] != "" &&
-		config[blockstorage.AzureClientSecret] != "")
+		config[blockstorage.AzureClientSecret] != ""
 }
 
 // determine if the combination of creds are MSI creds
 func isMSICredsAvailable(config map[string]string) bool {
 	_, clientIDok := config[blockstorage.AzureClientID]
-	return (clientIDok && config[blockstorage.AzureTenantID] == "" &&
-		config[blockstorage.AzureClientSecret] == "")
+	return clientIDok && config[blockstorage.AzureTenantID] == "" &&
+		config[blockstorage.AzureClientSecret] == ""
+}
+
+func isDefaultCredsAvailable(config map[string]string) bool {
+	_, clientIDok := config[blockstorage.AzureClientID]
+	_, tenantIDok := config[blockstorage.AzureTenantID]
+	_, clientSecretOk := config[blockstorage.AzureClientSecret]
+	return !clientIDok && !tenantIDok && !clientSecretOk
+}
```

Comment:
I'm not quite sure I under the intention of this check. Lack of these credentials does not necessary mean Default creds will be available

Reply:
Removed